sql/src/main/java/io/crate/expression/operator/CIDROperator.java

```diff
@@ -0,0 +1,158 @@
+        if (false == cidr.contains("/") || ipStr.contains("/")) {
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                "operands are incorrect, expected [ip, cidr], got [%s, %s]", ipStr, cidr));
+        }
+        try {
+            BigInteger ip = new BigInteger(1, InetAddress.getByName(ipStr).getAddress());
+            return extractRange(cidr, (start, end) -> start.compareTo(ip) <= 0 && ip.compareTo(end) < 0);
+        } catch (UnknownHostException uhe) {
+            throw new IllegalArgumentException(uhe);
+        }
+    }
+
+    public static Tuple<InetAddress, InetAddress> extractRange(String cidr) {
+        return extractRange(cidr, (start, end) -> new Tuple<>(fromBigIntegerIp(start), fromBigIntegerIp(end)));
+    }
+
+    @FunctionalInterface
+    private interface IpRangeConsumer<R> {
+        R accept(BigInteger ipStartRange, BigInteger ipEndRange);
+    }
+
+    private static <R> R extractRange(String cidr, IpRangeConsumer<R> rangeConsumer) {
+        if (null == cidr || false == cidr.contains("/")) {
+            throw new IllegalArgumentException(String.format(
+                Locale.ENGLISH, "operand [%s] must conform with CIDR notation", cidr));
+        }
+        Tuple<InetAddress, Integer> tup = InetAddresses.parseCidr(cidr);
+        InetAddress inetAddress = tup.v1();
+        BigInteger base = new BigInteger(1, inetAddress.getAddress());
+        BigInteger mask = createMask(inetAddress.getAddress().length, tup.v2());
+        BigInteger start = base.and(mask);
+        BigInteger end = start.add(mask.not());
+        return rangeConsumer.accept(start, end);
+    }
+
+    private static InetAddress fromBigIntegerIp(BigInteger ip) {
+        byte [] bytes = ip.toByteArray();
+        int size = Math.abs(bytes.length - IPV6_ADDRESS_LEN) <= 0 ? IPV6_ADDRESS_LEN : IPV4_ADDRESS_LEN;
+        byte[] ipBytes = new byte[size];
+        System.arraycopy(bytes, bytes.length - size, ipBytes, 0, size);
+        try {
+            return InetAddress.getByAddress(ipBytes);
+        } catch (UnknownHostException uhe) {
+            throw new IllegalArgumentException(uhe);
+        }
+    }
+
+    private static BigInteger createMask(int addressSizeInBytes, int prefixLength) {
+        ByteBuffer maskBuffer = ByteBuffer.allocate(addressSizeInBytes);
+        if (IPV4_ADDRESS_LEN == addressSizeInBytes) {
+            maskBuffer = maskBuffer.putInt(-1);
+        } else {
+            maskBuffer = maskBuffer.putLong(-1L).putLong(-1L);
+        }
+        return new BigInteger(1, maskBuffer.array()).not().shiftRight(prefixLength);
+    }
+
+    private static class CIDRResolver extends BaseFunctionResolver {
+
+        private final BiFunction<String, String, Boolean> operation;
+
+        private CIDRResolver(BiFunction<String, String, Boolean> operation) {
+            super(FuncParams.builder(Param.of(DataTypes.IP), Param.of(DataTypes.STRING)).build());
+            this.operation = operation;
```

Comment:
Might want to make this concrete for now, since there is only 1 operation used. 

Or create some kind of `BiScalar` (See `UnaryScalar`) and wrap it using `Scalar.withOperator`